json_encoder_bench_test.go

```diff
@@ -34,6 +34,18 @@ type logRecord struct {
 	Fields  map[string]interface{} `json:"fields"`
 }
 
+func BenchmarkJSONEncoderNest(b *testing.B) {
+	f := func(kv KeyValue) error {
+		kv.AddString("s", "string")
+		return nil
+	}
+	for i := 0; i < b.N; i++ {
+		enc := newJSONEncoder()
+		enc.Nest("nested", f)
+		enc.Free()
+	}
+}
```

Comment:
Good point, replaced with something that is more realistic by having an inline closure + use of a variable as the parameter (which actually makes it ~10ns slower).